app/controllers/concerns/jwt_validation.rb

```diff
@@ -0,0 +1,51 @@
+module JwtValidation
+  extend ActiveSupport::Concern
+
+  class UnsupportedIssuer < StandardError; end
+  class UnverifiedJWT < StandardError; end
+  class InvalidJWT < StandardError; end
+
+  included do
+    rescue_from InvalidJWT, with: :render_bad_request
+
+    rescue_from(
+      UnsupportedIssuer, UnverifiedJWT,
+      JSON::JWT::VerificationFailed, JSON::JWK::Set::KidNotFound,
+      OIDC::AccessPolicy::AccessError,
+      with: :render_not_found
+    )
+  end
+
+  def jwt_key_or_secret
+    raise NotImplementedError
+  end
+
+  def decode_jwt
+    @jwt = JSON::JWT.decode_compact_serialized(params.permit(:jwt).require(:jwt), jwt_key_or_secret)
```

Comment:
Is `:skip_verification` handled by the JWT lib?

Reply:
yes